### PR TITLE
feature(kensetsu): improve rate precision

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Improves stability fee rate precision.

This feature requires change on frontend, in `sora2-substrate-js-library/packages/util/src/kensetsu/index.ts`, function `updateCollateralInterestCoefficient`, function `now` returned value should be changed from ms to seconds (divide by 1000).

### CODE REVIEW

**Comment by Alexey-N-Chernyshov** [comment]
Let's keep it for the next release, our frontend is overloaded with tasks.

**Review by vovac12 — APPROVED** [review]

File: pallets/kensetsu/src/lib.rs [hunk 1]
```diff
@@ -2351,7 +2352,8 @@ pub mod pallet {
                                 risk_parameters: new_risk_parameters,
                                 total_collateral: Balance::zero(),
                                 stablecoin_supply: balance!(0),
-                                last_fee_update_time: Timestamp::<T>::get(),
+                                // now from ms to seconds, no need to check
+                                last_fee_update_time: Timestamp::<T>::get() / 1000u32.into(),
```
Comment (wer1st) [inline pallets/kensetsu/src/lib.rs:2356]: It should be checked operation

File: pallets/kensetsu/src/migrations.rs [hunk 2]
```diff
@@ -126,13 +127,19 @@ pub mod v1_to_v2 {
...
+                new_risk_parameters.stability_fee_rate = new_risk_parameters
+                    .stability_fee_rate
+                    .saturating_mul(FixedU128::from_u32(1000));
                 crate::CollateralInfo {
-                    risk_parameters: self.risk_parameters,
+                    risk_parameters: new_risk_parameters,
                     total_collateral: self.total_collateral,
                     stablecoin_supply: self.kusd_supply,
-                    last_fee_update_time: self.last_fee_update_time,
+                    last_fee_update_time: self.last_fee_update_time / 1000u32.into(),
```
Comment (wer1st) [inline pallets/kensetsu/src/migrations.rs:142]: Also should be checked

File: pallets/kensetsu/src/lib.rs (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply pallets/kensetsu/src/lib.rs:2356]: In case of overflow or `1000 == 0`? 😄

File: pallets/kensetsu/src/migrations.rs (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply pallets/kensetsu/src/migrations.rs:142]: No, there is no way to recover from error in migration.

Comment (wer1st) [reply pallets/kensetsu/src/migrations.rs:142]: At least log the error

Comment (wer1st) [reply pallets/kensetsu/src/migrations.rs:142]: Please, just log the error

**Review by shlavik — APPROVED** [review]

**Pull request merged**